### ISSUE TITLE
CSV Column for Scores that allows admins to see score discrepancies

### ISSUE THREAD

**Issue by Allicolyer** [Iridescent-CM/technovation-app#1951]
**Description**
Problem: Every year during judging, staff spends time calculating the range of scores for all submissions in a separate spreadsheet. You can see that work [here](https://docs.google.com/spreadsheets/d/1-CYXJEJwwIEW6WwYC-VrL_e78WMVKGFhagLPKXrnibw/edit#gid=824758563)

The purpose is to find submissions that have a big difference (more than 35 points) between their highest and lowest score, in other words, have a range that is greater than 35 points. This helps us find outliers in the scoring process and judges who might be scoring unfairly.

**Solution**
It would save us a lot of time if we could add a column to the score CSV report that already does this work for us. We can call this column "range".

**Describe alternatives you've considered**
I think it makes sense to calculate the range, or the difference between the highest and the lowest. 

However, with large ranges you don't always find an outlier, sometimes you just see a large spread of scores. We can also consider specifically looking for outliers, by calculating if one score is more than 17 points away from the average score. However, I think it's simpler and just as effective to look at the range.

**Acceptance Criteria**

1. As an admin
2. I can add a column to the CSV report called range
3. This column lists the difference between the highest and lowest score for each submission

**Comment by adrichey** [comment]
@ophelieh and @kmorton2017 I have a questions on this. I looked at the CSV export to see what the column data should look like. Can we elaborate a bit more on what the column, or potentially columns, should look like? Here are things I need insight on:
1. Should there be two columns for range, official and unofficial range? Or should official and unofficial be lumped into one range?
2. Additionally, we will need to display the range on multiple lines. Because we show each individual score for every team, we will be displaying the range on multiple lines. For example, if a team received four scores (73, 67, 9, and 7), the range would be 64 and each of those four scores in the CSV would show 64 as the range. I want to confirm that this is all right since we cannot reorganize this without changing the entire CSV structure or creating a new type of CSV export.

I am marking this as blocked until I get feedback on this.

**Comment by Allicolyer** [comment]
Hey @adrichey!

1. Let's only count official scores in the range 
2. It's okay for it to appear on multiple lines 

As long as we all know how this is working I think it will be okay.

**Comment by adrichey** [comment]
Sounds good! Thanks @Allicolyer. I will mark this as unblocked.

**Comment by adrichey** [comment]
This is now ready for testing. Thanks @stenington for knocking this out!

**Comment by ophelieh** [comment]
I selected "quarterfinal range" to add to the spreadsheet, however, once I downloaded it, it did not show. 

![uploads/cd82f678-3cea-4179-a102-882bc4984e0e/image.png](https://s3-us-west-2.amazonaws.com/prod.huboard.com/uploads%2Fcd82f678-3cea-4179-a102-882bc4984e0e%2Fimage.png)


Also, I chose to focus on 1 team and it is really weird, because that teams shows both as virtual and live. Not sure how that's possible. 


![uploads/d5ad6744-1578-4a4f-9ea1-f3a993447213/image.png](https://s3-us-west-2.amazonaws.com/prod.huboard.com/uploads%2Fd5ad6744-1578-4a4f-9ea1-f3a993447213%2Fimage.png)



Finally, would it be possible to add "score" for the "total" column (F). And add "average score" for column H? See below


![uploads/aa086f7e-a803-482c-b466-cac5007b8a94/image.png](https://s3-us-west-2.amazonaws.com/prod.huboard.com/uploads%2Faa086f7e-a803-482c-b466-cac5007b8a94%2Fimage.png)


Thank you!

**Comment by ophelieh** [comment]
@kmorton2017 Can you test today?

**Comment by stenington** [comment]
I'll take a look at the CSV export problems.

**Comment by stenington** [comment]
The scores page uses two totally separate datagrids for page display and CSV export, so that seems like a pre-existing bug to me prior to this ticket. I tried to do the quick-fix of simply having both use the same grid, and I get a 500 when I try to export so unfortunately it's not a quick fix.

I think we probably want to open a separate ticket asking for the page and export to use the same datagrid, and then we can continue tweaking columns if needed after that.

**Comment by stenington** [comment]
With a little more effort I figured out how I can make the export have the same columns as you see on screen. @ophelieh would that be a good fix here?

**Comment by ophelieh** [comment]
That would be great! :) I'm not sure I can see the range either on the platform though, I would need to recheck.

**Comment by stenington** [comment]
Okay, the CSV should now have the same columns shown on screen.

**Comment by ophelieh** [comment]
@kmorton2017 Can you help test that this week?

**Comment by ophelieh** [comment]
Looks great and works as expected. 

I'm going to be annoying and ask:
* In the more columns, would it be possible to add if it's a score from a live event or virtual? 
* *Katie* are you ok with just seeing the average score, or do we want to see all the scores for each team? @kmorton2017 

Thank you!

**Comment by stenington** [comment]
@ophelieh Each line of the results is a team instead of a single score, so I can't put if a score was live or virtual, but I think I could put whether or not that team was at an official RPE during quarterfinals. Would that work? 

I think I'm going to wait for @kmorton2017 to respond about scores though, because of we need to go back to showing one score per line instead of one team per line, I'll have a bunch more questions about the best way to accomplish that.

**Comment by ophelieh** [comment]
That would be great - to see if teams were at official RPE. :) 

And I think it would be fine to just see scores on 1 line, especially if it would cause more work. 

Thank you @stenington!

**Comment by stenington** [comment]
Okay, the scores page now has an optional "Quarterfinals official judging" column that tells you whether the virtual or live scores are being considered as the official scores for each team. It should only say "live" or teams that were at official RPEs.

<img width="541" alt="screen shot 2019-02-22 at 3 00 47 pm" src="https://user-images.githubusercontent.com/55319/53267964-c5ac6900-36b2-11e9-952d-65e3160fddbe.png">

**Comment by kmorton2017** [comment]
i am touching base with ophelie over slack to understand her question. ^^

**Comment by ophelieh** [comment]
Awesome. Thanks @stenington! @kmorton2017 Do you feel ok with this? If so, we can close this ticket.

**Comment by ophelieh** [comment]
Just tested it again. Looks good to go. Thank you @stenington
